components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1827,6 +1830,54 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         return subscription;
     }
 
+    protected async ensureIsUsageBasedFeatureFlagEnabled(user: User): Promise<void> {
+        const teams = await this.teamDB.findTeamsByUser(user.id);
+        const isUsageBasedBillingEnabled = await getExperimentsClient().getValueAsync(
```

Comment:
If feature flags don't work as expected in the `server`, I'd vote for not using them right now (a client-side feature flag is enough), and open a separate issue about fixing the server-side feature flags so that we can use them.

Reply:
@jankeromnes How about adding a pre-pr to make it work on server properly and then using it here? I'm happy to pair if that makes it easier. Boy/girl scout rules apply.

Reply:
@easyCZ That's the dashboard implementation, right?

The server-side implementation seems to be here: https://github.com/gitpod-io/gitpod/blob/main/components/server/src/experiments.ts

Reply:
Interesting, wasn't aware this landed (and my IDE resolved the method to the dashboard one). This must be stray from another PR. It also lives under the root of server/src which it should not.

This will in fact work for server. However, there's a second problem with it as it hard-codes a single Client ID which is for the preview/staging environment, but not production.

Let's land as is, and fix it up in a follow-up PR. We'll need to move the file, but also inject the Client ID based on environment we're running in. The current version would also break self-hosted environments as it reaches out to an external source.

Reply:
Follow-up issue: https://github.com/gitpod-io/gitpod/issues/10524